### PR TITLE
Revlist

### PR DESCRIPTION
**Pull request by dinosaure**
This is a test about the `rev-list` command. This computation compare what `git` wants to send to a server when we `push` and what `ocaml-git` wants to _PACK_ to send it to a server.

### CODE REVIEW

**Comment by dinosaure** [comment]
Propulsed by the #MirageOS hackathon and #Sun obviously.

**Comment by dinosaure** [comment]
About materials to test it, we use only the _usual_ PACK file because `git` segfault with the _bomb_ package ... And travis seems to be happy.

File: test/common/test_data.ml [hunk 1]
```diff
@@ -12,6 +12,18 @@ module type S = sig
   val create: Fpath.t -> (t, error) result Lwt.t
 end
 
+let output_of_command ?(env = [||]) ?(input = (fun _ -> ())) command =
+  let ic, oc, ec = Unix.open_process_full command env in
```
Comment (samoht) [inline test/common/test_data.ml]: `git` and associated tools are not the path here, so I am not sure how this is supposed to work.

File: test/git-unix/test_rev_list.ml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  let git_rev_list cs =
+    let command = Fmt.strf "git rev-list --objects --stdin" in
+    let input = fun oc ->
+      let ppf = Format.formatter_of_out_channel oc in
+      let pp_elt ppf = function
+        | `Negative hash -> Fmt.pf ppf "^%a" Store.Hash.pp hash
+        | `Object hash -> Store.Hash.pp ppf hash in
+      Fmt.(pf ppf) "%a\n%!" Fmt.(list ~sep:(const string "\n") pp_elt) (List.concat (List.map input_rev_list cs)) in
+    let git_dir = Fpath.(root / ".git") in
```
Comment (samoht) [inline test/git-unix/test_rev_list.ml]: Can't you pass `--git-dir <dir>` to your command instead of tweaking the environment?

**Comment by dinosaure** [comment]
Cleaned and ready to merge 👍 !

**Pull request merged**